### ISSUE TITLE
Change checks for unknown validators in Stellar relay pallet

### ISSUE THREAD

**Issue by ebma** [pendulum-chain/spacewalk#544]
## Context 

The stellar relay pallet currently rejects the whole 'proof' if one of the passed SCP envelopes was signed by an unknown validator or if any other related check fails, see [here](https://github.com/pendulum-chain/spacewalk/blob/d74009bf1d8c27fbd1f314c4fe486cd230e8b73f/pallets/stellar-relay/src/validation.rs#L163-L189). However, this check is a little too strict as it might happen that a set of envelopes can still be used to build a valid proof, even if some of the contained envelopes are 'invalid'.

## TODO

Instead of returning an error, we should rather change the logic so that the [validate_envelopes](https://github.com/pendulum-chain/spacewalk/blob/d74009bf1d8c27fbd1f314c4fe486cd230e8b73f/pallets/stellar-relay/src/validation.rs#L152) function checks each envelope for validity but only _returns the valid envelopes_. These valid envelopes can then be passed to the [check_for_valid_quorum_set](https://github.com/pendulum-chain/spacewalk/blob/d74009bf1d8c27fbd1f314c4fe486cd230e8b73f/pallets/stellar-relay/src/lib.rs#L589) function to continue with the existing validity checks.

**Comment by ebma** [comment]
@pendulum-chain/product this ticket loosens some requirements on the Stellar relay pallet. This is necessary because we are currently seeing some proofs failing on Amplitude which are likely to be fixed by this change. We will try to get this change included in the next runtime upgrade for polkadot-v1.1.0.

**Comment by prayagd** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Pendulum-6408a0365cda59c57d39bee5/issues/pendulum-chain/spacewalk/544?planning-poker) with Zenhub @b-yap @bogdanS98 @ebma @gianfra-t @TorstenStueber

**Comment by TorstenStueber** [comment]
@ebma do you have an example of such an "invalid" validator? Do we understand why that happens?

**Comment by TorstenStueber** [comment]
No, I was just curious. I wondered whether it is related to some partial Tier 1 quorum definition change again that we were not aware of.

Reasoning: [this pending ticket](https://github.com/pendulum-chain/tasks/issues/298#issuecomment-2168363585) about Tier 1 automation still requires more thought but I postponed that in favor of prioritizing the prototype. I wanted to understand whether it could become more important again.

**Comment by ebma** [comment]
I checked the slot in more detail again and realized that it's still the same problem that I documented already a while ago. The respective redeem request was created a very long time ago and thus the slot is very old too. And the archived messages for that slot still contain the old LOBSTR2 validator which is why the stellar-relay pallet cannot derive valid consensus since we already registered the new LOBSTR2 public key on chain. See [this](https://www.notion.so/satoshipay/23-12-18-Stellar-relay-unknown-validator-issue-522472011fee4187a6d64c2c1c007c11) Notion page.

**Comment by TorstenStueber** [comment]
Okay, then this issue here makes a lot of sense. Thanks!

**Comment by prayagd** [comment]
@b-yap moving this to development, as i see a PR linked. Please move back if not expected.
